src/main/java/com/pusher/client/channel/impl/PrivateEncryptedChannelImpl.java

```diff
@@ -124,6 +127,72 @@ public void updateState(ChannelState state) {
         }
     }
 
+    @Override
+    public PusherEvent prepareEvent(String event, String message) {
+        try {
+
+            Map receivedMessage = GSON.fromJson(message, Map.class);
+            final String decryptedMessage = decryptMessage((String) receivedMessage.get("data"));
+            receivedMessage.replace("data", decryptedMessage);
+
+            return GSON.fromJson(
+                    GSON.toJson(receivedMessage), PusherEvent.class);
```

Comment:
I'm not sure about this round-trip via GSON. If you look at the constructor for `PusherEvent`, it just wraps a map, so if we update the `"data"` key in our map, we should be able to construct a PusherEvent around it directly without needing to use serialisation.

Reply:
Hmm. I'll try again to see what I can do. It was proving to be a little difficult to get this working.

Reply:
Okay - so i'm passing in the Map<String, Object> into the PusherEvent and all the tests still pass. Thank you for the feedback!